Core/GDCore/Events/Builtin/CommentEvent.h

```diff
@@ -46,8 +46,8 @@ class GD_CORE_API CommentEvent : public gd::BaseEvent {
   const gd::String& GetComment() const { return com1; }
   void SetComment(const gd::String& comment) { com1 = comment; }
 
-  virtual std::vector<const gd::EventsList*> GetAllCommentsVectors() const;
-  virtual std::vector<gd::EventsList*> GetAllCommentsVectors();
+  virtual std::vector<const gd::String*> GetAllSearchableStrings(BaseEvent &event) const;
+  virtual std::vector<gd::String*> GetAllSearchableStrings( BaseEvent &event);

```

Comment:
Not sure here i guess there are mistakes in and after parenthesis.
I guess also one of them should return the vector of string ?
I didn't understand why two similar lines. with **Const** and without.

Reply:
As said in another comment: 

The one that is `const` will tell the compiler "please check that the object is not modified". It also tell the compiler "allow me to call this method on a const object". If you do:

```cpp
class A {
  void methodNonConst();
  void methodThatIsConst() const;
}
```

then you can do:
```cpp
void someOtherFunction(const A &a) { // Here I say "I take a reference to A, and I won't modify it"
  a.methodNonConst(); // NOT allowed, compiler will reject it. You're trying to modify the object it seems!
  a.methodThatIsConst(); // Fine! Nothing will be modified
}
```

Note that if you pass just a reference, there is no difference:
```cpp
void someOtherFunction(A &a) { // Here I say "I take a reference to A, and I maybe modify it"
  a.methodNonConst(); // Allowed. The object might be modified
  a.methodThatIsConst(); // Also allowed!
}
```

This is called "const-correctness": https://www.cprogramming.com/tutorial/const_correctness.html
(don't try to understand everything at first)

---

You also need only one method here (which can be const as not changing anything):

```
virtual std::vector<gd::String> GetAllSearchableStrings() const;
```